lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants
+  """
+  import Nx.Defn
+  alias __MODULE__
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and
+  sample targets `y`. Depending on number of classes the function chooses
+  either binary or multinomial logistic regression.
+  """
+  defn(fit(x, y, opts \\ []), do: fit_verify(x, y, opts))
```

Comment:
```suggestion
  defn fit(x, y, opts \\ []), do: fit_verify(x, y, opts)
```

Reply:
If this doesn't work, we need to fix the .formatter.exs to use import_deps like EXLA and Torchx do

Reply:
Indeed, the formatter adds these parentheses.